src/Middleware/AuthorizationMiddleware.php

```diff
@@ -111,13 +111,13 @@ public function __invoke(ServerRequestInterface $request, ResponseInterface $res
 
         try {
             $response = $next($request, $response);
+            if ($this->getConfig('requireAuthorizationCheck') && !$service->authorizationChecked()) {
+                throw new AuthorizationRequiredException(['url' => $request->getRequestTarget()]);
```

Comment:
Won't this result in a redirect when the redirect handler is used?

Reply:
Only when the handler is configured to catch this exception class. Redirect handler redirects only missing identities by default.

We could skip authentication in a catch block instead if accidental redirects are your concern.

Reply:
Good point. I didn't consider that.

Reply:
This change will also allow to catch `AuthorizationRequiredException` if someone wants to handle it in any special way.